File: deepseek/2.Estructura de Datos Pila (Stack)/c++/pila.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

int main() {
    stack<string> s;
    while (true) {
        cout << "¿Desea 'push', 'pop' o 'salir'? ";
        string action;
        cin >> action;
        if (action == "salir") {
            break;
        } else if (action == "push") {
            cout << "Ingrese el valor a apilar: ";
            string item;
            cin >> item;
            s.push(item);
        } else if (action == "pop") {
            if (!s.empty()) {
                cout << "Elemento desapilado: " << s.top() << endl;
                s.pop();
            } else {
                cout << "La pila está vacía" << endl;
            }
        }
    }
    return 0;
}

File: deepseek/2.Estructura de Datos Pila (Stack)/ruby/pila.rb
class Stack
  def initialize
    @items = []
  end

  def push(item)
    @items.push(item)
  end

  def pop
    @items.pop unless is_empty?
  end

  def is_empty?
    @items.empty?
  end
end

# Interacción con el usuario
stack = Stack.new
loop do
  print "¿Desea 'push', 'pop' o 'salir'? "
  action = gets.chomp.downcase
  case action
  when "salir"
    break
  when "push"
    print "Ingrese el valor a apilar: "
    item = gets.chomp
    stack.push(item)
  when "pop"
    popped = stack.pop
    puts popped ? "Elemento desapilado: #{popped}" : "La pila está vacía"
  end
end